__tests__/Auth0Client.test.ts

```diff
@@ -512,4 +512,33 @@ describe('Auth0Client', () => {
       value: originalUserAgent
     });
   });
+
+  it('uses the cache for multiple token requests with audience and scope', async () => {
```

Comment:
Just a general question; do we want to migrate tests out of `index.test.ts` into this file? I'm conscious that we're starting to develop a set of tests across two places for the same functions.

Perhaps we should dedicate some effort to unifying the tests somewhere to make them more discoverable.

Reply:
Agreed